ksql-engine/src/test/java/io/confluent/ksql/engine/KsqlEngineTest.java

```diff
@@ -670,12 +670,48 @@ public void shouldCleanUpInternalTopicsOnClose() {
     query.start();
 
     // When:
-    query.close();
+    query.close(true);
 
     // Then:
     verify(topicClient).deleteInternalTopics(query.getQueryApplicationId());
   }
 
+  @Test
+  public void shouldNotCleanUpInternalTopicsOnCloseIfCleanFalse() {
+    // Given:
+    final QueryMetadata query = KsqlEngineTestUtil.executeQuery(
+        serviceContext,
+        ksqlEngine,
+        "select * from test1 EMIT CHANGES;",
+        KSQL_CONFIG, Collections.emptyMap()
+    );
+    query.start();
+
+    // When:
+    query.close(false);
+
+    // Then:
+    verifyNoMoreInteractions(topicClient);
+  }
+
+  @Test
+  public void shouldNotCleanUpInternalTopicsOnEngineClose() {
+    // Given:
+    final QueryMetadata query = KsqlEngineTestUtil.executeQuery(
+        serviceContext,
+        ksqlEngine,
+        "select * from test1 EMIT CHANGES;",
+        KSQL_CONFIG, Collections.emptyMap()
+    );
+    query.start();
+
+    // When:
+    ksqlEngine.close();
+
+    // Then:
+    verifyNoMoreInteractions(topicClient);
+  }
```

Comment:
This is a transient query, which should have its internal topics deleted when the engine closes.  Can we update the functionality and test to reflect this?

Also, can we then add another similar test that shows persistent queries do _not_ have their internal state deleted.